shell/platform/linux/fl_engine.cc

```diff
@@ -8,19 +8,108 @@
 
 #include <gmodule.h>
 
+static constexpr int kMicrosecondsPerNanosecond = 1000;
+
+// Unique number associated with platform tasks
+static constexpr size_t kPlatformTaskRunnerIdentifier = 1;
+
 struct _FlEngine {
   GObject parent_instance;
 
+  // Thread the GLib main loop is running on
+  GThread* thread;
+
   FlDartProject* project;
   FlRenderer* renderer;
   FLUTTER_API_SYMBOL(FlutterEngine) engine;
+
+  // PlatformMessageHandler keyed by channel name
+  GHashTable* platform_message_handlers;
+
+  // Messages being processed
+  GPtrArray* message_handles;
```

Comment:
What's the purpose of this array? I don't see any use of it other than adding and removing things. Is it just lifetime enforcement? If so, doesn't providing the handle to the handler take care of that?

Reply:
It was to keep ownership of the GBytes and the handle, but I agree, with further thought it's not really necessary. It does require the API user to take a reference to the GBytes, but that's probably not unreasonable.

I have been struggling with the way to pass the response handle to the FlEngine user - it seems unnecessary to wrap it inside a structure with one element. But the only way we can otherwise pass it is using a void pointer, which is ugly.